src/main.c

```diff
@@ -142,10 +148,117 @@ static bool is_valid_stream(char *s)
 	return true;
 }
 
+char* get_saved_file_path(void)
+{
+	char* home_path = getenv("XDG_STATE_HOME");
+
+	if (home_path == NULL)
+	{
+		// fallback to $HOME/.local/state
+		home_path = getenv("HOME");
```

Comment:
I would still check `home_path` here and react accordingly.

Reply:
Do you mean to check `home_path` if `$HOME` is not set?

Also, would it better if instead of exiting on error (e.g `home_path` not set at all) user will get notification
like "Can't save the game"/"Can't open game save..."?

Reply:
I mean to check `home_path` for NULL (for whatever reason).
Yes I think it would be better to just exit the function/tell the user that he cant save the save/open the game.